### PR TITLE
[fix]: experimental import fix

### LINKED ISSUE #1042: [Defect in 0.4.7] pygit2/pgzip not automatically installed.
Good morning! Thank you for publishing another hotfix for `fairscale` 🙇‍♂️ I'm sure this will be well received.

# Bug description
Due to the inclusion of entry_points the `pygit2/pgzip` packages are now required by fairscale even though its experimental and optional.

```
entry_points={"console_scripts": ["wgit = fairscale.experimental.wgit.__main__:main"]},
```
https://github.com/facebookresearch/fairscale/blob/main/setup.py#L72

# Steps to reproduce
(in a virtualenv)
```
$ pip install fairscale==0.4.7
$ python -c "import fairscale"
...
Error: please pip install pygit2 module to use wgit
```

# Expected behavior
(in a virtualenv)
```
$ pip install fairscale==0.4.7
$ python -c "import fairscale"
$ # no error
```

# Workaround
(in a virtualenv)
```
$ pip install fairscale!=0.4.7
$ python -c "import fairscale"
$ # no error
```

Would the solution here be to add `pygit2` & `pgzip` to requirements.txt or optionally import them in https://github.com/facebookresearch/fairscale/blob/main/fairscale/experimental/wgit/__init__.py ?

### PR DESCRIPTION
**Pull request by min-xu-ai**
## What does this PR do?
fixing #1042 

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/main/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/main/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by min-xu-ai** [comment]
@anupambhatnagar and @riohib, this looks good?

File: fairscale/experimental/__init__.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 from typing import List
 
-from . import nn, optim, wgit
+# Don't import sub-modules to avoid experimental stuff gets imported directly
+# when user does an `import fairscale`
```
Comment (min-xu-ai) [inline fairscale/experimental/__init__.py:13]:
```suggestion
# when user does an `import fairscale`. This can cause experimental code's import
# dependencies (like pygit2) to be leaked into the fairscale main code.
```

File: fairscale/experimental/__init__.py [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 from typing import List
 
-from . import nn, optim, wgit
+# Don't import sub-modules to avoid experimental stuff gets imported directly
```
Comment (anupambhatnagar) [inline fairscale/experimental/__init__.py:12]: Suggesting changing `Don't import sub-modules to avoid experimental stuff` to `Don't import sub-modules as experimental stuff`

**Review by anupambhatnagar — COMMENTED** [review]
I would rephrase it to:

Don't import sub-modules as experimental stuff gets imported directly
when user does an `import fairscale`

**Review by anupambhatnagar — APPROVED** [review]
👍
